src/Altinn.App.Core/Features/Correspondence/Builder/CorrespondenceRequestBuilder.cs

```diff
@@ -0,0 +1,299 @@
+        }
+    }
+}
+
+/// <summary>
+/// Builder factory for creating <see cref="CorrespondenceRequest"/> objects
+/// </summary>
+public class CorrespondenceRequestBuilder : CorrespondenceBuilderBase, ICorrespondenceRequestBuilder
+{
+    private string? _resourceId;
+    private OrganisationNumber? _sender;
+    private string? _sendersReference;
+    private CorrespondenceContent? _content;
+    private DateTimeOffset? _allowSystemDeleteAfter;
+    private DateTimeOffset? _dueDateTime;
+    private IReadOnlyList<OrganisationOrPersonIdentifier>? _recipients;
+    private DateTimeOffset? _requestedPublishTime;
+    private string? _messageSender;
+    private IReadOnlyList<CorrespondenceExternalReference>? _externalReferences;
+    private IReadOnlyDictionary<string, string>? _propertyList;
+    private IReadOnlyList<CorrespondenceReplyOption>? _replyOptions;
+    private CorrespondenceNotification? _notification;
+    private bool? _ignoreReservation;
+    private IReadOnlyList<Guid>? _existingAttachments;
+
+    private CorrespondenceRequestBuilder() { }
+
+    /// <summary>
+    /// Creates a new <see cref="CorrespondenceRequestBuilder"/> instance
+    /// </summary>
+    public static ICorrespondenceRequestBuilderResourceId Create() => new CorrespondenceRequestBuilder();
+
+    /// <inheritdoc/>
+    public ICorrespondenceRequestBuilderSender WithResourceId(string resourceId)
+    {
+        NotNullOrEmpty(resourceId, "Resource ID cannot be empty");
+        _resourceId = resourceId;
+        return this;
+    }
+
+    /// <inheritdoc/>
+    public ICorrespondenceRequestBuilderSendersReference WithSender(OrganisationNumber sender)
+    {
+        NotNullOrEmpty(sender, "Sender cannot be empty");
+        _sender = sender;
+        return this;
+    }
+
+    /// <inheritdoc/>
+    public ICorrespondenceRequestBuilderSendersReference WithSender(string sender)
+    {
+        NotNullOrEmpty(sender, "Sender cannot be empty");
+        _sender = OrganisationNumber.Parse(sender);
+        return this;
+    }
+
+    /// <inheritdoc/>
+    public ICorrespondenceRequestBuilderRecipients WithSendersReference(string sendersReference)
+    {
+        NotNullOrEmpty(sendersReference, "Senders reference cannot be empty");
+        _sendersReference = sendersReference;
+        return this;
+    }
+
+    /// <inheritdoc/>
+    public ICorrespondenceRequestBuilderAllowSystemDeleteAfter WithRecipient(OrganisationOrPersonIdentifier recipient)
+    {
+        return WithRecipients([recipient]);
+    }
+
+    /// <inheritdoc/>
+    public ICorrespondenceRequestBuilderAllowSystemDeleteAfter WithRecipient(string recipient)
+    {
+        return WithRecipients([recipient]);
+    }
+
+    /// <inheritdoc/>
+    public ICorrespondenceRequestBuilderAllowSystemDeleteAfter WithRecipients(
+        IReadOnlyList<OrganisationOrPersonIdentifier> recipients
+    )
+    {
+        _recipients = [.. _recipients ?? [], .. recipients];
```

Comment:
Sanity-checking again, since we use a mutable model here (for the builder in general), why do we not just add to the existing list?

Reply:
No reason that I can think of. This was just the pattern that stuck from the first POC we did a while back.

You prefer to init the list of required, then just add the new items?

Reply:
Right, if we get the same behavior but better performance then it is tempting

Reply:
The exception to this rewrite was `CorrespondenceContent.Attachments`, which is stored as a `IReadOnlyList` in the model. Due to the nature of _not_ building this from the `CorrespondenceContentAttachmentBuilder`, but instead from the main `CorrespondenceRequestBuilder`, this particular list had to be instantiated on each modification (unless I tracked it locally in the builder, which felt messy): https://github.com/Altinn/app-lib-dotnet/pull/897/files#diff-115e9f775e8a2658fc9cd7c0afd11678d96d4496ccae4ed2b5373131b1cd33dbR291